malariagen_data/ag3.py

```diff
@@ -314,6 +320,482 @@ def _repr_html_(self):
+        if region.end:
+            slice_stop = region.end
+        else:
+            slice_stop = None
+        loc_region = slice(slice_start, slice_stop)
+
+        return d[loc_region]
+
+    def genome_features(
+        self, region=None, attributes=("ID", "Parent", "Name", "description")
+    ):
+        """Access genome feature annotations.
+
+        Parameters
+        ----------
+        region: str or list of str or Region or list of Region
+            Contig name (e.g., "2L"), gene name (e.g., "AGAP007280"), genomic
+            region defined with coordinates (e.g., "2L:44989425-44998059") or a
+            named tuple with genomic location `Region(contig, start, end)`.
+            Multiple values can be provided as a list, in which case data will
+            be concatenated, e.g., ["3R", "3L"].
+        attributes : list of str, optional
+            Attribute keys to unpack into columns. Provide "*" to unpack all
+            attributes.
+
+        Returns
+        -------
+        df : pandas.DataFrame
+            A dataframe of genome annotations, one row per feature.
+
+        """
+        debug = self._log.debug
+
+        if region is None:
+            df = super().genome_features(region=region, attributes=attributes)
+            return df
+
+        elif region is not None:
+            region = self.resolve_region(region)
+            debug("normalise to list to simplify concatenation logic")
+            if isinstance(region, Region):
+                region = [region]
+
+            if all([r.contig in self.contigs for r in region]):
+                df = super().genome_features(region=region, attributes=attributes)
+                return df
+
+            elif all([r.contig in ["2RL", "3RL"] for r in region]):
+                parts = []
+                for r in region:
+                    debug("concatenating right and left arm genome features")
+                    contig_r, contig_l = (
+                        r.contig[0] + r.contig[1],
+                        r.contig[0] + r.contig[2],
+                    )
+                    df_r = super().genome_features(
+                        region=contig_r, attributes=attributes
+                    )
+                    df_l = super().genome_features(
+                        region=contig_l, attributes=attributes
+                    )
+                    max_r = df_r.query("type == 'chromosome'").loc[0, "end"]
+                    df_l = df_l.assign(
+                        start=lambda x: x.start + max_r, end=lambda x: x.end + max_r
+                    )
+                    df_part = pd.concat([df_r, df_l], axis=0)
+
+                    debug("apply region query")
+                    if r.end is not None:
+                        df_part = df_part.query(f"start <= {r.end}")
+                    if r.start is not None:
+                        df_part = df_part.query(f"end >= {r.start}")
+                    parts.append(df_part)
```

Comment:
Seemingly small point, but this code is duplicated from the superclass implementation.

It would be nice to avoid code duplication wherever possible, even if small. (Should be possible to avoid any code duplication.)